### PR TITLE
Add TLS configuration for PostgreSQL

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Add TLS configuration for PostgreSQL

**Why?**
Improve security

**How did you test it?**
Make PostgreSQL use TLS
Make Temporal server use TLS

* Create keys / certs
```zsh
openssl genrsa -aes256 -out ca-key.pem 4096
openssl req -new -x509 -days 365 -key ca-key.pem -sha256 -out ca.pem

openssl genrsa -out server-key.pem 4096
openssl req -subj "/CN=localhost" -sha256 -new -key server-key.pem -out server.csr
openssl x509 -req -days 365 -sha256 -in server.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out server-cert.pem

openssl genrsa -out client-key.pem 4096
openssl req -subj '/CN=localhost' -new -key client-key.pem -out client.csr
openssl x509 -req -days 365 -sha256 -in client.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out client-cert.pem

rm -v client.csr server.csr

chmod -v 0400 ca-key.pem server-key.pem client-key.pem 
chmod -v 0444 ca.pem server-cert.pem client-cert.pem
```

* Config PostgreSQL 
pg_hba.conf: 
```zsh
hostssl    all             all             127.0.0.1/32            md5 clientcert=1                                                      
hostssl    all             all             ::1/128            md5 clientcert=1     
```
postgresql.conf:
```zsh
ssl = on
ssl_cert_file = '<path to PostgreSQL cert>'
ssl_key_file = '<path to PostgreSQL key>'
ssl_ca_file = '<path to PostgreSQL CA>'
```

* Config Temporal server
```yaml
datastores:
    postgres-default:
      sql:
        pluginName: "postgres"
        databaseName: "temporal"
        connectAddr: "127.0.0.1:5432"
        connectProtocol: "tcp"
        user: "temporal"
        password: "temporal"
        maxConns: 20
        maxIdleConns: 20
        maxConnLifetime: "1h"
        tls:
          enabled: true
          certFile: "<path to server cert>"
          keyFile: "<path to server key>"
          caFile: "<path to server CA>"
    postgres-visibility:
      sql:
        pluginName: "postgres"
        databaseName: "temporal_visibility"
        connectAddr: "127.0.0.1:5432"
        connectProtocol: "tcp"
        user: "temporal"
        password: "temporal"
        maxConns: 2
        maxIdleConns: 2
        maxConnLifetime: "1h"
        tls:
          enabled: true
          certFile: "<path to server cert>"
          keyFile: "<path to server key>"
          caFile: "<path to server CA>"
```

**Potential risks**
N/A

### CODE REVIEW

File: common/persistence/sql/sqlplugin/mysql/tls.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	host, _, err := net.SplitHostPort(cfg.ConnectAddr)
+	if err != nil {
+		return fmt.Errorf("error in host port from ConnectAddr: %v", err)
+	}
+
+	// TODO: create a way to set MinVersion and CipherSuites via cfg.
+	tlsConfig := auth.NewTLSConfigForServer(host)
+
+	if cfg.TLS.CaFile != "" {
```
Comment (underrun) [inline common/persistence/sql/sqlplugin/mysql/tls.go:53]: Possibly also support `CaData` like we do for cassandra

Comment (wxing1292) [reply common/persistence/sql/sqlplugin/mysql/tls.go:53]: one thing at a time.

this only try to make mysql / postgresql similar in terms of file structure, not changing anything for mysql

**Review by sergeybykov — APPROVED** [review]

File: common/persistence/sql/sqlplugin/mysql/plugin.go [hunk 2]
```diff
@@ -47,7 +41,7 @@ import (
 const (
 	// PluginName is the name of the plugin
 	PluginName                   = "mysql"
-	dsnFmt                       = "%s:%s@%v(%v)/%s"
+	dsnFmt                       = "%v:%v@%v(%v)/%v"
```
Comment (alexshtin) [inline common/persistence/sql/sqlplugin/mysql/plugin.go:44]: what is the point of having this as separate const?

Comment (wxing1292) [reply common/persistence/sql/sqlplugin/mysql/plugin.go:44]: do you mean this?
```go
const dsnFmt = "%v:%v@%v(%v)/%v"
```

usually, all string const are defined at the very top

**Pull request merged**